rpc/proto/mayastor.proto

```diff
@@ -368,3 +371,38 @@ message BdevUri {
 message CreateReply {
   string name = 1;
 }
+
+message BlockDevice {
+  message Partition {
+    string parent = 1;          // devname of parent device to which this partition belongs
+    uint32 number = 2;          // partition number
+    string name = 3;            // partition name
+    string scheme = 4;          // partition scheme: gpt, dos, ...
+    string typeid = 5;          // partition type identifier
+    string uuid = 6;            // UUID identifying partition
+  }
+  message Filesystem {
+    string fstype = 1;          // filesystem type: ext3, ntfs, ...
+    string label = 2;           // volume label
+    string uuid = 3;            // UUID identifying the volume (filesystem)
+    string mountpoint = 4;      // path where filesystem is currently mounted
+  }
```

Comment:
is there a reason why these structs are defined nested? It seems that if one needs to refer to the struct name, he/she has to do BlockDevice.Filesystem instead of just Filesystem. That's probably ok. Though we don't do that anywhere else so just asking what is the intent here.

Reply:
The Partition struct contains information pertaining *only* to partitions and likewise Filesystem contains *only* information related to the file system. In many cases one or both of these will contain no data - and indeed rust compiles these to Option<Filesystem>, etc . In our use case in particular, we are mainly interested in the case when this object has a value "None". In fact this is precisely one of the criteria for deciding whether or not the device is available for use.